File: 0017 - Letter Combinations of a Phone Number/solution.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        m = {'2':'abc','3':'def','4':'ghi','5':'jkl',
            '6':'mno','7':'pqrs','8':'tuv','9':'wxyz'}
        res = [''] if digits else []
        for i in digits:
            res = [p+q for p in res for q in m[i]]
        return res

File: 0017 - Letter Combinations of a Phone Number/solution.cpp
class Solution {
public:
    vector<string> letterCombinations(string digits) {
        if(digits.size() == 0) return {};
        vector<string> m{"","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"};
        vector<string> res{""}, tmp;
        for(auto d : digits){
            for(auto s : res){
                for(auto c : m[d-'0']){
                    tmp.push_back(s+c);
                }
            }
            res.swap(tmp);
            tmp.clear();
        }
        return res;
    }
};